Fix openapi3.referencedDocumentPath

kshlm opened the pull request:
`openapi3.referencedDocument` tried to `path.Join` two URLs, and then
generate a new `url.URL` by parsing the obtained path. The URL
obtained at the end would be invalid, when the base path is HTTP URL.

Given base path as 'http://example.com/schemas' and ref as '/schema1',
the obtained end URL would be 'http:/example.com/schemas'. This happens
because `path.Join` cleans the path it generates and replaces repeated
'//' with a single '/'.

This is fixed by doing a `path.Join` on the base URL.Path and the ref,
and setting this result as the URL.Path of the new URL.

fenollp commented:
Hi there, thanks for the patch!
Could you add a few tests on this function? Especially one that would not pass before your fix.
Thanks

kshlm commented:
I've added the tests.

I'm not a 100% sure on the about the expected result of `referencedDocumentPath`. I would have expected it to be `path.Dir(basePath)+refPath`. But other tests seem to expect it to be `path.Dir(basePath)+path.Dir(refPath)+"/"`. Which is what I've implemented the changes and the tests for.

fenollp commented on openapi3/swagger_loader_referenced_document_path_test.go:
```diff
@@ -0,0 +1,63 @@
+
+func TestReferencedDocumentPath(t *testing.T) {
+	httpURL, err := url.Parse("http://example.com/path/to/schemas/test1.yaml")
+	if err != nil {
```
Please call `require.NoError(t, err)` instead.

fenollp commented on openapi3/swagger_loader_referenced_document_path_test.go:
```diff
@@ -0,0 +1,63 @@
+	if err != nil {
+		panic(err)
+	}
+	fileURL, err := url.Parse("path/to/schemas/test1.yaml")
```
Please call `require.NoError(t, err)` instead.

fenollp commented on openapi3/swagger_loader_referenced_document_path_test.go:
```diff
@@ -0,0 +1,63 @@
+		},
+	} {
+		result, err := referencedDocumentPath(test.path, test.ref)
+		require.NotNil(t, result)
```
Remove this line.

fenollp commented on openapi3/swagger_loader_referenced_document_path_test.go:
```diff
@@ -0,0 +1,63 @@
+	} {
+		result, err := referencedDocumentPath(test.path, test.ref)
+		require.NotNil(t, result)
+		require.Nil(t, err)
```
Please call `require.NoError(t, err)` instead.

fenollp reviewed (approved):
Thanks for the tests :) I have a few nitpicks but after that it'll be merged.

kshlm commented:
Not sure why tests are failing. They're unrelated to my change in the `openapi3` package.
